govcd/edgegateway.go

```diff
@@ -311,87 +339,172 @@ func (eGW *EdgeGateway) AddSNATRule(networkHref, externalIP, internalIP, descrip
 		return nil, fmt.Errorf("%s", combinedTaskErrorMessage(task.Task, err))
 	}
 
+	var createdNatRule *types.NatRule
+
 	err = eGW.Refresh()
 	if err != nil {
 		return nil, fmt.Errorf("error refreshing edge gateway: %#v", err)
 	}
 
-	return eGW.EdgeGateway.Configuration.EdgeGatewayServiceConfiguration.NatService.NatRule, nil
+	for _, natRule := range eGW.EdgeGateway.Configuration.EdgeGatewayServiceConfiguration.NatService.NatRule {
+		if natRule.Description == mappingId {
+			createdNatRule = natRule
+			break
+		}
+	}
+
+	if createdNatRule == nil {
+		return nil, fmt.Errorf("error creating SNAT rule, didn't match created rule")
+	}
+
+	createdNatRule.Description = description
+
+	return eGW.UpdateNatRule(createdNatRule)
+}
+
+// getPseudoUuid creates unique ID/UUID
```

Comment:
Please add a Unit test for this function. BTW, could we use a library function (such as Google UUID) for this purpose?

Reply:
Could you help me with test? I didn't want to introduce dependency for simple matter for generating ID which is used for 1s. Even math.rand may be sufficient..

Reply:
Sent to you in Slack

Reply:
Thank you, added The test